File: Data Structures & Algorithms/binary-search/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
        int right = nums.length - 1;
        int left = 0;
        int middle = nums.length/2;
        while (right >= left){
            if (target == nums[middle]){
                return middle;
            }
            if (target > nums[middle]){
                left = middle + 1;
            }
            else {
                right = middle - 1;
            }
            middle = (right + left) / 2;
            
        }
        
        return -1;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-2.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        left, right = 0, len(matrix) * len(matrix[0]) - 1
        while left <= right:
            mid = (right - left) // 2 + left
            row = mid // len(matrix[0])
            col = mid % len(matrix[0])
            print(left, mid, right)
            print(row,col)
            if matrix[row][col] == target:
                return True
            if matrix[row][col] < target:
                left = mid + 1
            else:
                right = mid - 1
        return False